### ISSUE TITLE
Upgrade web to react 18

### ISSUE THREAD

**Issue by chxash** [shapeshift/web#1739]
### Task is not blocked anymore by chakra-ui release, but since it was released very recently, proceed with caution

React 18 was released on march (https://reactjs.org/blog/2022/03/29/react-v18.html).
Upgrading the web app to react 18 will enable the use of new features, like automatic batching and concurrent rendering.
This issue is about performing the **upgrade to react 18 in the web app**.

Some notable points in order to do it :
- update `react`, `react-dom`, `@types/react`, `@types/react-dom` and any other required dependencies in package.json
- update `framer-motion` to at least `6.2.10` (https://github.com/framer/motion/issues/1509)
- update chakra-ui to v2 (https://chakra-ui.com/guides/migration)
- update index.tsx
- basically follow the upgrade guides given below
- fix any compilation errors that could arise

Application should work well without any regression.
This task will probably require a full anti-regression testing after implementation.

### References and additional details

ReactJS blog post about upgrading to react 18 : https://reactjs.org/blog/2022/03/08/react-18-upgrade-guide.html
Another useful link for upgrading : https://blog.logrocket.com/upgrading-react-18-typescript/
Tool to automatically fix some compilation errors : https://github.com/eps1lon/types-react-codemod

### Acceptance Criteria

Application should be working well with 0 regression
Unit tests are green
e2e tests are green

### Estimated effort

Multiple days

**Comment by 0xean** [comment]
@chxash - do you want to take this one on?

**Comment by chxash** [comment]
@0xean I would like to take it but I am well occupied with fox page work right now ! If noone takes it by the time I have some bandwith I will gladly take it

**Comment by 0xean** [comment]
low priority.
